crates/common/types/transaction.rs

```diff
@@ -1400,7 +1366,16 @@ impl PrivilegedL2Transaction {
         let nonce = u256_nonce.to_big_endian();
 
         Some(keccak_hash::keccak(
-            [to.as_bytes(), &value, &nonce].concat(),
+            [
+                to.as_bytes(),
+                &value,
+                &nonce,
+                self.recipient.as_bytes(),
+                self.recipient.as_bytes(),
+                &U256::from(self.gas_limit).to_big_endian(),
```

Comment:
`gas_limit` is not in the function docs. Don't know which is the correct one, but should be the same.

Edit: accordingly to the contract, I think `gas_limit` is correct, so we should add it to the comment above